eval-adxl355-pmdz: Add support for new trigger structure

rbolboac opened the pull request:
- Adapt adxl355 driver and eval-adxl355-pmdz project for new trigger structure.
- Add IIO_TRIGGER_EXAMPLE support for maxim platform
- Minor code refactoring

rbolboac commented:
Can be merged only after https://github.com/analogdevicesinc/no-OS/pull/1363 is merged.

rbolboac commented:
- master rebase

rbolboac commented:
- master rebasse

amiclaus commented on projects/eval-adxl355-pmdz/src/platform/stm32/platform_src.mk:
```diff
@@ -1,14 +1,14 @@
 INCS += $(PLATFORM_DRIVERS)/stm32_delay.h     \
-		$(PLATFORM_DRIVERS)/stm32_gpio.h      \
```
as far as i can see in other projects, the *.mk files are using tabs for indentation. so not sure if this change is required.

amiclaus commented on projects/eval-adxl355-pmdz/src/common/common_data.c:
```diff
@@ -46,44 +46,48 @@
-struct no_os_uart_init_param uip = {
+struct no_os_uart_init_param adxl355_uart_ip = {
 	.device_id = UART_DEVICE_ID,
-	.baud_rate = 115200,
```
is this change related to the commit description? looks like this commit does more than adapting to the new trigger structure.

rbolboac replied:
no, this change should go to the code refactoring part. I will update this. thanks!

amiclaus reviewed (commented):
some minor comments. Although, I think the adapt to new trigger struct commits are not quite doing only that. Moreover, the changes in the driver and project with respect to this subject (new trigger structure) are interdependent, therefore they should be in the same commit, otherwise the driver changes commit will break the project at that specific point in commit history.

rbolboac commented:
> some minor comments. Although, I think the adapt to new trigger struct commits are not quite doing only that. Moreover, the changes in the driver and project with respect to this subject (new trigger structure) are interdependent, therefore they should be in the same commit, otherwise the driver changes commit will break the project at that specific point in commit history.

I fixed the changes related to the new trigger structured and I added all of them in a single commit such that the build is successful for each commit and the project is working as expected.

buha reviewed (approved):
looks good to me
